skimage/io/sift.py

```diff
@@ -62,9 +60,10 @@ def _sift_read(f, mode='SIFT'):
 
     data = np.fromfile(f, sep=' ')
     if data.size != nr_features * datatype.itemsize / np.dtype(float).itemsize:
-        raise IOError("Invalid %s feature file." % mode)
+        raise IOError("Invalid {} feature file.".format(mode))
 
-    f.close()
+    if f_is_str:
+        f.close()
```

Comment:
I don't think so. I think @soupault is correct. If it's a string, the socket is opened by this function, and we want to close it before ending. However, if it's a socket, you don't want to close it because the user might want to use it afterwards.